devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy

```diff
@@ -0,0 +1,314 @@
+	// log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(basicv1.BasicReport cmd) {
+	log.debug "rx $cmd"
+	handleDeviceEvent(cmd.value)
+}
+
+def zwaveEvent(sensorbinaryv2.SensorBinaryReport cmd) {
+	log.debug "rx $cmd"
+	def result = []
+
+	if (cmd.sensorType == sensorbinaryv2.SensorBinaryReport.SENSOR_TYPE_TAMPER) {
+		result << createEvent(name: "tamper", value: "detected")
+	} else {
+		result = handleDeviceEvent(cmd.sensorValue)
+	}
+	result
+}
+
+def zwaveEvent(notificationv3.NotificationReport cmd) {
+	def result = []
+
+	log.debug "rx $cmd"
+
+	if (cmd.notificationType == notificationv3.NotificationReport.NOTIFICATION_TYPE_BURGLAR) {
+		if (cmd.event == 3) {
+			result << createEvent(name: "tamper", value: "detected")
+		}
+	} else {
+		log.warn "Unknown cmd.notificationType: ${cmd.notificationType}"
+		result << createEvent(descriptionText: cmd.toString(), isStateChange: false)
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Unhandled Z-Wave command $cmd"
+	createEvent(displayed: false, descriptionText: "$device.displayName: $cmd")
+}
+
+def handleDeviceEvent(value) {
+	log.debug "handleDeviceEvent $value"
+	def result = [
+		createEvent([name: "switch", value: value ? "on" : "off", displayed: false]),
+		createEvent([name: "alarm", value: value ? "both" : "off"])
+	]
+
+	// value != 0 doesn't necessarily trigger the below events,
+	// but value == 0 clears them
+	if (!value) {
+		result << createEvent([name: "chime", value: "off"])
+		result << createEvent([name: "tamper", value: "clear"])
+	}
+
+	result
+}
+
+def generateCommand(command) {
+	def sound = (command && soundMap.containsKey(command)) ? soundMap[command] : soundMap[defaultSound]
+	log.debug "Sending $command"
+
+	secure(zwave.notificationV3.notificationReport(notificationType: sound.notificationType, event: sound.event))
+}
+
+def chime() {
+	log.debug "chime!"
+
+	// Chime is kind of special as the alarm treats it as momentary
+	// and thus sends no updates to us, so we'll send this and request an update
+	sendEvent(name: "chime", value: "chime")
+
+	[
+		generateCommand("Chime"),
+		secure(zwave.basicV1.basicGet())
+	]
+}
+
+def on() {
+	log.debug "sending on"
+	generateCommand(state.sound)
+}
+
+def off() {
+	log.debug "sending off"
+
+	secure(zwave.basicV1.basicSet(value: 0x00))
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def test() {
+	[
+		secure(zwave.basicV1.basicSet(value: 0xFF)),
+		"delay 3000",
+		secure(zwave.basicV1.basicSet(value: 0x00))
+	]
+}
+
+private secure(physicalgraph.zwave.Command cmd) {
+	def zwInfo = zwaveInfo
+	// This model is explicitly secure, so if it paired "the old way" and zwaveInfo doesn't exist then encapsulate
+	if (!zwInfo || (zwInfo?.zw?.contains("s") && (cmd.commandClassId == 0x20 || zwInfo.sec?.contains(String.format("%02X", cmd.commandClassId))))) {
```

Comment:
not a big deal, but I think `0x20` should fall under security command list.

Reply:
Yeah, I think so, actually. I copied this from the Aeon Siren in my z-wave cert PR, and now that you mention it I think this was because the Aeon devices would ignore insecure on that command class but it wasn't listed in the secure list. :/